electron/js/menu/TrayIconHandler.js

```diff
@@ -52,15 +52,14 @@ class TrayIconHandler {
     return this.platform.IS_WINDOWS ? 'ico' : 'png';
   }
 
-  get hasTrayMenuSupport() {
-    return !this.platform.IS_MAC_OS;
-  }
-
   get hasOverlaySupport() {
     return this.platform.IS_WINDOWS;
   }
 
   updateBadgeIcon(win, count) {
+    if (!this.appIcon) {
```

Comment:
This is not specific to Windows. The `flashFrame` and `setBadgeCount` also need to be called on macOS